docker/development/medusa-be/Dockerfile

```
@@ -2,17 +2,81 @@ FROM node:22-slim AS base
+
+# Remove all apps/libs not needed by medusa-be, clean caches
+RUN find apps -maxdepth 1 -mindepth 1 -type d ! -name 'medusa-be' -exec rm -rf {} + && \
+    rm -rf libs && \
+    rm -rf node_modules apps/medusa-be/node_modules apps/medusa-be/.medusa
+
+RUN pnpm install --filter=medusa-be...
+
+# Inline dummy secrets satisfy Medusa's build-time validation without embedding real secrets in image layers.
+# Real secrets must be provided at runtime via env vars, or Medusa will crash (by design).
+ARG FEATURE_PPL_ENABLED=0
+RUN NODE_ENV=production \
```

Hadolint (2.14.0)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)